### PR TITLE
Make it possible to update vasypd update-interval

### PR DESCRIPTION
**Pull request by erobhal**
We need the possibility to update vasypd update-interval in vas.conf on our vas-nis proxies. This patch makes it possible-

### CODE REVIEW

**Comment by Phil-Friderici** [comment]
@erobhal congrats for fixing the gems that are causing the spec test to fail ! My fix is using puppetlabs_spec_helper 2.0.2 as last Ruby 1.8.7 compatible version instead.

Will add some inline comments regarding your changed functionality.

File: manifests/init.pp [hunk 1]
```diff
@@ -31,6 +31,7 @@
   $domain_change                                        = false,
   $sitenameoverride                                     = 'UNSET',
   $vas_conf_client_addrs                                = 'UNSET',
+  $vas_conf_vasypd_update_interval                      = 'USE_DEFAULTS',
```
Comment (Phil-Friderici) [inline manifests/init.pp]: USE_DEFAULTS is mostly used to be able to specify different default values depending on other details.
I don't see this here, so please just set the default to the original value of '1800'. A simple integer validation of $vas_conf_vasypd_update_interval is good enough afterwards.

Comment (Phil-Friderici) [inline manifests/init.pp]: Would be good to have support to convert stringified integers to real integers.

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -406,6 +407,16 @@
...
+    context 'with vas_conf_vasypd_update_interval set to invalid string (non-integer)' do
+      let :params do
+        { :vas_conf_vasypd_update_interval => '600invalid' }
+      end
+
+      it 'should fail' do
+        expect { should contain_class('vas') }.to raise_error(Puppet::Error, /validate_integer/)
+      end
+    end
+
```
Comment (Phil-Friderici) [inline spec/classes/init_spec.rb]: Please move the validation check to the 'variable type and content validations' section instead (at the bottom).
Add a new block like this to test support of integers and stringified integers:
```
      'integer' => {
        :name    => %w(vas_conf_vasypd_update_interval),
        :valid   => [242,'242'],
        :invalid => ['string', %w(array), { 'ha' => 'sh' }, 2.42, true, false, nil],
        :message => 'Expected.*to be an Integer',
      },
```

Please also add an atomic test that only tests your new parameter and nothing else. Like the test for vdi_mode for example.

File: templates/vas.conf.erb [hunk 3]
```diff
@@ -38,7 +38,7 @@
 [vasypd]
  search-base = <%= @nismaps_ou %>
  split-groups = true
- update-interval = 1800
+ update-interval = <%= @vas_conf_vasypd_update_interval_real %>
```
Comment (Phil-Friderici) [inline templates/vas.conf.erb]: After $vas_conf_vasypd_update_interval has been converted to an real integer, please use $vas_conf_vasypd_update_interva_integer.

**Comment by erobhal** [comment]
Updated with enhancement requests

File: manifests/init.pp [hunk 4]
```diff
@@ -129,6 +130,9 @@
   }
 
   # validate params
+  validate_integer($vas_conf_vasypd_update_interval)
+  $vas_conf_vasypd_update_interval_integer = 0 + $vas_conf_vasypd_update_interval # str2integer
```
Comment (Phil-Friderici) [inline manifests/init.pp]: Unsure if this 'trick' will work in the future.
Usually I recommend to use stdlib's floor() instead [1]. See ssh module for an example [2]

[1] https://github.com/puppetlabs/puppetlabs-stdlib#floor
[2] https://github.com/ghoneycutt/puppet-module-ssh/blob/v3.49.1/manifests/init.pp#L472

Comment (Phil-Friderici) [inline manifests/init.pp]: When done, please remove the gem related changes and squash the functional commits into one so that I can merge them fast :)

Comment (erobhal) [inline manifests/init.pp]: Ok, I'll change it. Though the 0 + trick is actually the documented method [1]. :-)

[1] [https://docs.puppet.com/puppet/4.9/lang_data_number.html#converting-strings-to-numbers](url)

Comment (Phil-Friderici) [inline manifests/init.pp]: Than please never mind, just squash the functionals into one please.

**Comment by erobhal** [comment]
Changes done to make Travis CI green has been removed. Functional commits squashed into one.

**Comment by Phil-Friderici** [comment]
@erobhal Released in v0.21.0. Thank you !

**Pull request merged**